### ISSUE TITLE
Nuitka is not changing __debug__ to False when --python-flag=-O is used.

### ISSUE THREAD

**Issue by jcrmatos** [Nuitka/Nuitka#222]
Hello,

Nuitka is not changing `__debug__` to False when --python-flag=-O is used.

```
(.venv) C:\Users\JMatos\OneDrive\Python\uniclave>python -m nuitka --version
0.6.0.6
Python: 3.7.1 (v3.7.1:260ec2c36a, Oct 20 2018, 14:05:16) [MSC v.1915 32 bit (Int
el)]
Executable: C:\Users\JMatos\OneDrive\Python\uniclave\.venv\Scripts\python.exe
OS: Windows
Arch: x86
```

Thanks,

JM

**Comment by kayhayen** [comment]
I cannot reproduce this issue, but there is a difference with `-OO` there:

Simply printing `__debug__` in Python2

`
$ python -m nuitka --run Mini.py 
True
$ python -m nuitka --run --python-flag=-O Mini.py  
False
$ python -m nuitka --run --python-flag=-OO Mini.py 
True

**Comment by kayhayen** [comment]
Nuitka normally complains about unsupported values.

I pushed to the factory branch a fix for this issue, or least what I am guessing it is. Now for -OO it will also be false. Correct me if I am confused and it is indeed about -O,

**Comment by jcrmatos** [comment]
Hello,

Your are correct. The problem is occurring with --python-flag=-O.

I was trying a simple test (using the command line arguments your showed) and got a strange error.

I created a test.py with only a single line
`print(__debug__)`

When I run

```
python test.py
True
python -O test.py
False
python -OO test.py
False
```

Everything is working fine.

If I try to run one of these
python -m nuitka --run test.py
python -m nuitka --run --python-flag=-O test.py
python -m nuitka --run --python-flag=-O test.py

Python( or Nuitka) complains that

```
(.venv) C:\Users\JMatos\OneDrive\Python>python -m nuitka --run teste1.py
Fatal Python error: initfsencoding: unable to load the file system codec
ModuleNotFoundError: No module named 'encodings'

Current thread 0x00001734 (most recent call first):

(.venv) C:\Users\JMatos\OneDrive\Python>
```

Thanks,

JM

**Comment by jcrmatos** [comment]
Hello,

Found the solution for the  strange error above.
Had to create the env var PYTHONHOME and point it to Python interpreter. Don't know why this is needed when running with nuitka, but it solved the error.
Now the tests:
```
C:\Users\JMatos\OneDrive\Python>python -m nuitka --run test.py
True

C:\Users\JMatos\OneDrive\Python>python -m nuitka --run --python-flag=-O test.py
True
```

The second one should have returned False.

Thanks,

JM

**Comment by kayhayen** [comment]
So 0.6.1 somehow fixed this I think, I saw that 0.6.0.6 was indeed affected, but it's old news now, and 0.6.1 does the expected thing.

Anyway, obviously for `-OO` the fix was still to be done, but then it wasn't really supported anyway.
